src/auth/session/storage/redis.ts

```diff
@@ -71,6 +71,32 @@ export class RedisSessionStorage implements SessionStorage {
     return true;
   }
 
+  public async deleteSessions(ids: string[]): Promise<boolean> {
+    await this.ready;
+    for (const id of ids) {
+      await this.deleteSession(id);
+    }
+    return true;
+  }
+
+  public async findSessionsByShop(
+    shop: string,
+  ): Promise<SessionInterface[] | {[key: string]: unknown}[] | undefined> {
+    await this.ready;
+
+    const keys = await this.client.keys('*');
```

Comment:
I wonder if it's worth storing the ids for specific shops under a single key, so that we don't have to iterate over every entry to find the ids for a shop. Quite a big change, but I think it's worth it.

Reply:
K, that would be better (anything would be better than this current search implementation!) but since it's such a big change to the underlying data structure, I'll leave it for its own PR.